ui/plugin-system/src/runtime/time-series-queries.ts

```diff
@@ -140,3 +144,33 @@ function useTimeSeriesQueryContext(): TimeSeriesQueryContext {
     refreshKey,
   };
 }
+
+/**
+ * Get active time series queries for query results summary
+ */
+export function useActiveTimeSeriesQueries() {
+  const queryClient = useQueryClient();
+  const queryCache = queryClient.getQueryCache();
+  const timeSeriesQueries = getActiveTimeSeriesQueries(queryCache);
+  return timeSeriesQueries;
+}
+
+/**
+ * Filter all cached queries down to only active time series queries
+ */
+export function getActiveTimeSeriesQueries(cache: QueryCache) {
+  const queries = cache
+    .findAll({ type: 'active' })
+    .filter((query) => {
+      const firstPart = query.queryKey?.[0] as UnknownSpec;
+      if (firstPart?.kind) {
+        return (firstPart?.kind as string).startsWith(TIME_SERIES_QUERY_KEY);
+      }
+      return false;
+    })
+    .filter((query) => query.isActive)
+    .map((query) => {
```

Comment:
Just a general thought here: this is a lot of loops happening with the multiple `filter`/`map`/etc. calls. Wonder if it would be simpler/faster to just write this as a single for...of loop?

Reply:
Yeah good call, I [refactored this](https://github.com/perses/perses/pull/1019/commits/42566aa6983981768ed9da33ca2f73c37fe7eb9c) so it's much simpler/faster now